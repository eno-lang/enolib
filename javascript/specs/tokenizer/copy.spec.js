const { inspectTokenization } = require('./util.js');

const input = `
template:

key < template
key    < template
     key     < template
`.trim();

describe('Copy tokenization', () => {
  it('performs as specified', () => {
    expect(inspectTokenization(input)).toMatchSnapshot();
  });
});
