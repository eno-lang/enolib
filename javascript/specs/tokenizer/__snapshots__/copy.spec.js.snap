// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy tokenization performs as specified 1`] = `
Object {
  "_document": Object {
    "depth": 0,
    "elements": Array [
      Object {
        "key": "template",
        "line": 0,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            8,
            9,
          ],
          "key": Array [
            0,
            8,
          ],
          "line": Array [
            0,
            9,
          ],
        },
        "type": Symbol(Empty Element),
      },
      Object {
        "key": "key",
        "line": 2,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            15,
            16,
          ],
          "key": Array [
            11,
            14,
          ],
          "line": Array [
            11,
            25,
          ],
          "template": Array [
            17,
            25,
          ],
        },
        "template": "template",
        "type": Symbol(Empty Element),
      },
      Object {
        "key": "key",
        "line": 3,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            33,
            34,
          ],
          "key": Array [
            26,
            29,
          ],
          "line": Array [
            26,
            43,
          ],
          "template": Array [
            35,
            43,
          ],
        },
        "template": "template",
        "type": Symbol(Empty Element),
      },
      Object {
        "key": "key",
        "line": 4,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            57,
            58,
          ],
          "key": Array [
            49,
            52,
          ],
          "line": Array [
            44,
            67,
          ],
          "template": Array [
            59,
            67,
          ],
        },
        "template": "template",
        "type": Symbol(Empty Element),
      },
    ],
    "type": Symbol(Document),
  },
  "_lineCount": 5,
  "_meta": Array [],
}
`;
